btree/btree.c

```diff
@@ -4530,16 +4534,21 @@ static int64_t btree_get_kv_tick(struct m0_sm_op *smop)
 						    P_SETUP);
 			}
 
-			oi->i_key_found = node_find(&node_slot,
-						    &bop->bo_rec.r_key);
-			lev->l_idx = node_slot.s_idx;
+			if (bop->bo_opc == M0_BO_GET) {
+				oi->i_key_found = node_find(&s,
+							    &bop->bo_rec.r_key);
+				lev->l_idx = s.s_idx;
+			} else
+				s.s_idx = bop->bo_opc == M0_BO_MINKEY ?  0 :
+					  node_count(s.s_node);
```

Comment:
Can we move these lines below at line 4546 so that the checks are minimal ?

Reply:
This won't make much difference. Have made the changes as per comment.